lib/galaxy/config/sample/job_conf.xml.sample_advanced

```diff
@@ -629,6 +635,18 @@
                argument by default. You can turn this off by setting singularity_cleanenv to `false`.
           -->
           <!-- <param id="singularity_cleanenv">true</param> -->
+          <!-- Singularity by default inherits the PID namespace, this can give
+               issues with multiprocessing, hence galaxy passes the `pid` argument
+               by default to isolate the PID namespace. You can turn this of by setting
+               singularity_pid to `false`.
+          -->
+          <!-- <param id="singularity_pid">true</param> -->
+          <!-- Singularity by default inherits the IPC namespace, this can give
+               issues with multiprocessing, hence galaxy passes the `ipc` argument
+               by default to isolate the PID namespace. You can turn this of by setting
+               singularity_ipc to `false`.
+          -->
+          <!-- <param id="singularity_ipc">true</param> -->
```

Comment:
Do we need to expose these 2 arguments as options, or should we just always enabled them as you are proposing to do for `--contain` ?

Reply:
`--contain` is not an invasive param. You can undo its effects by simply manually adding the volumes. 
This is not the case with `--ipc` and `--pid` (as far as I can see). So having an option to disable the flags might be very useful if they cause problems in some tools. 

We have never experienced problems with `--ipc` and `--pid` at our institute, in fact, we experienced problems when they were not used. However I still take the conservative approach here to allow admins to at least turn it off if it gives them problems.